src/API/Google/AdsCampaign.php

```diff
@@ -481,4 +484,21 @@ protected function parse_campaign_id( string $name ): int {
 			throw new Exception( __( 'Invalid campaign ID', 'google-listings-and-ads' ) );
 		}
 	}
+
+	/**
+	 * Convert location ID from a geo target constant resource name to an int.
+	 *
+	 * @param string $geo_target_constant Resource name containing ID number.
+	 *
+	 * @return int
+	 * @throws Exception When unable to parse resource ID.
+	 */
+	protected function parse_geo_target_location_id( string $geo_target_constant ): int {
+		try {
+			$parts = GeoTargetConstantServiceClient::parseName( $geo_target_constant );
+			return absint( $parts['criterion_id'] );
+		} catch ( ValidationException $e ) {
+			throw new Exception( __( 'Invalid geo target location ID', 'google-listings-and-ads' ) );
```

Comment:
Just double checking if this is intentional or if you think it's better to skip the location. Since there is no try/catch around when we call this function, one bad location will prevent any campaigns from being retrieved.
That might be intentional though since the Google API shouldn't be returning invalid locations (in that case it's good to leave as is).

Reply:
Yeah, it's intentional but skipping the location also makes sense to me 🤔 I kept it as is and added a test case to catch the exception in c7436c8 per your suggestion.